### PR TITLE
Refactor kubeadm config list-images

### PR DESCRIPTION
**Pull request by chuckha**
In preparation for creating a `kubeadm config images pull`
this commit refactors `kubeadm config list-images` into
`kubeadm config images list`.

Signed-off-by: Chuck Ha <ha.chuck@gmail.com>

**What this PR does / why we need it**:
Prepares the `kubeadm config images` subcommand so we can add more functionality to it.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Related to kubernetes/kubeadm#812

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/kubeadm/app/cmd/config.go [hunk 1]
```diff
@@ -206,50 +206,60 @@ func uploadConfiguration(client clientset.Interface, cfgPath string, defaultcfg
 	return uploadconfig.UploadConfiguration(internalcfg, client)
 }
 
-// NewCmdConfigListImages returns the "kubeadm images" command
-func NewCmdConfigListImages(out io.Writer) *cobra.Command {
+func NewCmdConfigImages(out io.Writer) *cobra.Command {
```
Comment (luxas) [inline cmd/kubeadm/app/cmd/config.go]: nit: add Godoc comment

File: cmd/kubeadm/app/cmd/config.go [hunk 2]
```diff
@@ -206,50 +206,60 @@ func uploadConfiguration(client clientset.Interface, cfgPath string, defaultcfg
...
+	cmd := &cobra.Command{
+		Use:   "images",
+		Short: "Interact with images required for a kubeadm init.",
```
Comment (luxas) [inline cmd/kubeadm/app/cmd/config.go]: suggest: `Interact with container images used by kubeadm.`

File: cmd/kubeadm/app/cmd/config.go [hunk 3]
```diff
@@ -206,50 +206,60 @@ func uploadConfiguration(client clientset.Interface, cfgPath string, defaultcfg
...
-			listImages, err := NewListImages(cfgPath, cfg)
+			imagesList, err := NewImagesList(cfgPath, cfg)
 			kubeadmutil.CheckErr(err)
-			kubeadmutil.CheckErr(listImages.Run(out))
+			kubeadmutil.CheckErr(imagesList.Run(out))
 		},
 	}
-	AddListImagesConfigFlag(cmd.PersistentFlags(), cfg, &featureGatesString)
-	AddListImagesFlags(cmd.PersistentFlags(), &cfgPath)
+	AddImagesListConfigFlag(cmd.PersistentFlags(), cfg, &featureGatesString)
```
Comment (luxas) [inline cmd/kubeadm/app/cmd/config.go]: plural like everything else? AddImagesListConfigFlags

**Review by luxas — COMMENTED** [review]
minor nits
/approve

please fix the small nits and then we can merge :+1:

**Review by luxas — APPROVED** [review]
/lgtm
/approve

**Comment by chuckha** [comment]
oops, that failure that ends up with a 500 is actually my failure to run the updated docs script. Fixing.

**Review by luxas — APPROVED** [review]
/lgtm
/approve

**Comment by chuckha** [comment]
lol sorry @luxas. I was struggling with figuring out the docs stuff. 🙈

**Review by timothysc — APPROVED** [review]
/lgtm

**Pull request merged**